gef-ui/server/controllers/application-details/index.test.js

```diff
@@ -323,6 +328,101 @@ describe('controllers/application-details', () => {
+          'partials/application-preview.njk',
+          expect.objectContaining({
+            canIssuedFacilitiesBeAmended: false,
+          }),
+        );
+      });
+
+      it(`renders 'application-preview' with canIssuedFacilitiesBeAmended=false when the submission type is ${DEAL_SUBMISSION_TYPE.MIA}`, async () => {
+        jest.mocked(isPortalFacilityAmendmentsFeatureFlagEnabled).mockReturnValueOnce(true);
+
+        api.getFacilities.mockResolvedValue(MOCKS.MockFacilityResponseNotChangedIssued);
+        mockApplicationResponse.status = DEAL_STATUS.UKEF_ACKNOWLEDGED;
+        mockApplicationResponse.submissionType = DEAL_SUBMISSION_TYPE.MIA;
+        api.getApplication.mockResolvedValueOnce(mockApplicationResponse);
+
+        await applicationDetails(mockRequest, mockResponse);
+
+        expect(mockResponse.render).toHaveBeenCalledWith(
+          'partials/application-preview.njk',
+          expect.objectContaining({
+            canIssuedFacilitiesBeAmended: false,
+          }),
+        );
+      });
+
+      it(`renders 'application-preview' with canIssuedFacilitiesBeAmended=false when the user type is not ${ROLES.MAKER}`, async () => {
+        jest.mocked(isPortalFacilityAmendmentsFeatureFlagEnabled).mockReturnValueOnce(true);
+
+        api.getFacilities.mockResolvedValue(MOCKS.MockFacilityResponseNotChangedIssued);
+        mockApplicationResponse.status = DEAL_STATUS.UKEF_ACKNOWLEDGED;
+        mockApplicationResponse.submissionType = DEAL_SUBMISSION_TYPE.AIN;
+        api.getApplication.mockResolvedValueOnce(mockApplicationResponse);
+
+        const mockCheckerRequest = MOCKS.MockRequestChecker();
+        mockCheckerRequest.flash = mockSuccessfulFlashResponse();
+
+        await applicationDetails(mockCheckerRequest, mockResponse);
+
+        expect(mockResponse.render).toHaveBeenCalledWith(
+          'partials/application-preview.njk',
+          expect.objectContaining({
+            canIssuedFacilitiesBeAmended: false,
+          }),
+        );
+      });
+
+      it(`renders 'application-preview' with canIssuedFacilitiesBeAmended=false when deal cancellation is not enabled`, async () => {
+        jest.mocked(isPortalFacilityAmendmentsFeatureFlagEnabled).mockReturnValueOnce(false);
+
+        api.getFacilities.mockResolvedValue(MOCKS.MockFacilityResponseNotChangedIssued);
+        jest.mocked(isPortalFacilityAmendmentsFeatureFlagEnabled).mockReturnValueOnce(true);
+
+        mockApplicationResponse.status = DEAL_STATUS.UKEF_ACKNOWLEDGED;
+        mockApplicationResponse.submissionType = DEAL_SUBMISSION_TYPE.AIN;
+        api.getApplication.mockResolvedValueOnce(mockApplicationResponse);
+
+        await applicationDetails(mockRequest, mockResponse);
+
+        expect(mockResponse.render).toHaveBeenCalledWith(
+          'partials/application-preview.njk',
+          expect.objectContaining({
+            canIssuedFacilitiesBeAmended: false,
+          }),
+        );
+      });
+
```

Comment:
Not a blocker for this PR, but we should really look at splitting this test up, it's absolutely massive and difficult to follow. Just over 600 lines in one file

Reply:
Yep, one for future👍 Will resolve here in the meantime